tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Support/Assertions.cs

```diff
@@ -45,9 +55,66 @@ public void AssertWorkflowInstanceMatchesExpectedWorkflow(WorkflowInstance workf
             }
         }
 
+        public void AssertInputArtifacts(TaskObject workflowRevisionTask, string payloadId, TaskExecution workflowInstanceTask)
+        {
+            foreach (var workflowArtifact in workflowRevisionTask.Artifacts.Input)
+            {
+                if (workflowArtifact.Value == "{{ context.input.dicom }}")
+                {
+                    workflowInstanceTask.InputArtifacts[workflowArtifact.Name].Should().Match($"{payloadId}/dcm/");
+                }
+            }
+        }
+
+        public void AssertInputArtifactsForTaskDispatch(TaskObject workflowRevisionTask, string payloadId, List<Messaging.Common.Storage> taskDispatchInput)
+        {
+            foreach (var workflowArtifact in workflowRevisionTask.Artifacts.Input)
+            {
+                if (workflowArtifact.Value == "{{ context.input.dicom }}")
+                {
+                    var taskDispatchArtifact = taskDispatchInput.FirstOrDefault(x => x.Name.Equals(workflowArtifact.Name));
+                    taskDispatchArtifact.RelativeRootPath.Should().Match($"{payloadId}/dcm/");
+                    taskDispatchArtifact.Bucket.Should().Match(TestExecutionConfig.MinioConfig.Bucket);
+                }
+            }
+        }
+
+        public void AssertOutputDirectory(TaskExecution task, string payloadId, string workflowInstanceId)
+        {
+            task.OutputDirectory.Should().Match($"{payloadId}/workflows/{workflowInstanceId}/{task.ExecutionId}");
+        }
+
+        public void AssertOutputArtifactsForTaskUpdate(Dictionary<string, string> workflowInstanceTaskOutput, List<Messaging.Common.Storage> taskUpdateOutput)
+        {
+            foreach (var artifact in taskUpdateOutput)
+            {
+                var workflowInstanceTaskArtifact = workflowInstanceTaskOutput.FirstOrDefault(x => x.Key.Equals(artifact.Name));
```

Comment:
Maybe .First(...) might be a better fit here?

Reply:
Updated